libr/core/cmd_anal.c

```diff
@@ -5077,6 +5075,42 @@ static void cmd_anal_aad(RCore *core, const char *input) {
 	r_list_free (list);
 }
 
+
+static bool archIsArmOrThumb(RCore *core) {
+	if (r_str_startswith (r_config_get (core->config, "asm.arch"), "arm")) {
+		ut64 bits = r_config_get_i (core->config, "asm.bits");
+		if (bits < 64) {
+			return true;
+		}
+	}
+	return false;
+}
+
+void _CbInRangeAav(RCore *core, ut64 from, ut64 to, int vsize, bool asterisk, int count) {
+	bool isarm = archIsArmOrThumb (core);
+	if (isarm) {
+		if (to & 1) {
+			// .dword 0x000080b9 in reality is 0x000080b8
+			to--;
+			r_anal_hint_set_bits (core->anal, to, 16);
+			// can we assume is gonna be always a function?
+		} else {
+			r_core_seek_archbits (core, from);
+			ut64 bits = r_config_get_i (core->config, "asm.bits");
+			r_anal_hint_set_bits (core->anal, from, bits);
+		}
+	}
+	if (asterisk) {
+		r_cons_printf ("ax 0x%"PFMT64x " 0x%"PFMT64x "\n", to, from);
+		r_cons_printf ("Cd %d @ 0x%"PFMT64x "\n", vsize, from);
+		r_cons_printf ("f fcn.0x%08"PFMT64x " @0x%08"PFMT64x, to, to);
+	} else {
+		r_core_cmdf (core, "ax 0x%"PFMT64x " 0x%"PFMT64x, to, from);
+		r_core_cmdf (core, "Cd %d @ 0x%"PFMT64x, vsize, from);
+		r_core_cmdf (core, "f fcn.0x%08"PFMT64x " @0x%08"PFMT64x, to, to);
```

Comment:
use = instead of @, because that's faster (avoid temporary seek + temporary block read)